lib/language_pack/ruby.rb

```diff
@@ -512,10 +534,15 @@ def install_ruby
     instrument 'ruby.install_ruby' do
       return false unless ruby_version
       installer = LanguagePack::Installers::RubyInstaller.installer(ruby_version).new(@stack)
+      file = "#{ruby_version.version_for_download}.tgz"
 
       if ruby_version.build?
         installer.fetch_unpack(ruby_version, build_ruby_path, true)
+        file.sub!("ruby", "ruby-build")
       end
+      @ruby_download_check = LanguagePack::Helpers::DownloadPresence.new(file)
```

Comment:
I'm not quite sure what the "build ruby" is (vs the normal ruby), but it seems like if `installer.fetch_unpack()` above were to fail (because the version doesn't exist), then `fetch_untar` (called by `fetch_unpack`) would raise `FetchError` and thus this line would never get called prior to the usage of `@ruby_download_check` in the `rescue`?

Reply:
Great catch. I fixed that. I also added a hatchet integration test for the scenario where there's a bad ruby version.